### ISSUE TITLE
Fix S6934 FP/FN: Child classes of RouteAttribute

### ISSUE THREAD

**Issue by JohnYoungers** [SonarSource/sonar-dotnet#8999]
### Description

For the new S6934 rule, instead of checking if the controller has `RouteAttribute` explicitly (https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/src/SonarAnalyzer.CSharp/Rules/SpecifyRouteAttribute.cs#L46), can we instead check if it has an attribute that inherits `RouteAttribute`?

### Repro steps

```
[AttributeUsage(AttributeTargets.Class)]
public sealed class CustomRouteAttribute : RouteAttribute
{
	public CustomRouteAttribute(string template)
		: base(template)
	{
	}
}

[CustomRoute("api/[controller]")]
public class ControllerWithInheritedRoute : Controller
{
    [HttpGet("Test")]
    public string Index() => "Hi!";
}
```

### Expected behavior

S6934 should not be triggered

**Comment by mary-georgiou-sonarsource** [comment]
Hello, @JohnYoungers.
Thanks a lot for reporting this, I confirm it's an FP.
However, I'll close this issue as a duplicate. Please follow #8985 instead.

**Comment by JohnYoungers** [comment]
thank you! looks like I was late to the party
